### PR TITLE
CT-2303 Added a feature of uploading the requests files

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to add a feature for London users to upload new request files after the FOI/SAR case is create

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2303

### CODE REVIEW

File: app/views/cases/attachments/new.html.slim [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      .column-two-thirds
+        p.form-hint
+          = t('cases.attachments.upload_hint_copy')
+    .grid-row
+      section.case-main
+        = render partial: 'shared/dropzone_form',
+              locals: { s3_direct_post: @s3_direct_post,
+                        file_input_name: 'uploaded_request_files[]' }
+        .form-group
+          label.form_label
```
Comment (digitalali-moj) [inline app/views/cases/attachments/new.html.slim:33]: Does this automatically add "for" attribute for the textarea?

File: app/views/cases/attachments/new.html.slim [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+= form_for @case, as: :case_attachment, url: case_attachments_path(@case), method: :post do |f|
+  .form-group
+    .grid-row
+      .column-two-thirds
+        p.form-hint
+          = t('cases.attachments.upload_hint_copy')
+    .grid-row
+      section.case-main
+        = render partial: 'shared/dropzone_form',
```
Comment (digitalali-moj) [inline app/views/cases/attachments/new.html.slim:29]: Is this the same upload form used everywhere or are there others, just asking as we have some keyboard accessibility issues. If it's the same we can fix that later as there is a ticket for it.

File: app/controllers/cases/attachments_controller.rb [hunk 3]
```diff
@@ -27,6 +29,34 @@ def destroy
...
+        @case = @case.decorate
+        flash.now[:alert] = service.error_message
+        render :new
+      else
+        if service.result == :ok
+          flash[:notice] = t('notices.request_uploaded')
+        end
+        redirect_to case_path @case
+      end
+    end
```
Comment (Polsonby) [inline app/controllers/cases/attachments_controller.rb:58]: Wondering if the inner conditional on line 53 ought to be on the same level as the outer conditional from line 47? 

Or is it like this intentionally to handle the `:blank` case from the service? 🤔 

Just looks a little unusual - I think normally for these we do 

```
    if service.result == :ok
      happy things
    else
      render or redirect with error
    end
```

File: config/locales/page_titles.en.yml [hunk 4]
```diff
@@ -42,3 +42,4 @@ en:
     stats: Performance Reports - Track-a-query
     upload_responses_page: Uploading a response - %{case_number} - Track-a-query
     move_team_page: Moving a business unit - %{team_name} - Track-a-query
+    add_request_attachment_page: Upload new files for this request
```
Comment (Polsonby) [inline config/locales/page_titles.en.yml:45]: Sorry, nitpick! Could we add the missing newline here?

**Review by Polsonby — COMMENTED** [review]
LGTM 👍 😁 - one question, one nitpick - happy to approve though

File: app/views/cases/attachments/new.html.slim (hunk 2 above)
Comment (ymao2) [reply app/views/cases/attachments/new.html.slim:29]: @digitalali-moj  yes, it is the same upload form used everywhere

File: app/views/cases/attachments/new.html.slim (hunk 1 above)
Comment (ymao2) [reply app/views/cases/attachments/new.html.slim:33]: I don't know, let me check

Comment (ymao2) [reply app/views/cases/attachments/new.html.slim]: no 'for' is added, I added it

File: app/controllers/cases/attachments_controller.rb (hunk 3 above)
Comment (ymao2) [reply app/controllers/cases/attachments_controller.rb:61]: I changed it to case statement with extra part for handling ':blank' value

File: app/views/cases/attachments/new.html.slim (hunk 1 above)
Comment (digitalali-moj) [reply app/views/cases/attachments/new.html.slim]: Is it matching the ID of the textarea? I can't see the ID here so hard to tell.

Comment (ymao2) [reply app/views/cases/attachments/new.html.slim]: 🤦‍♀️ ,   no, sorry! I corrected it "again", should it be right this time? 😳 
screenshot below
Screenshot 2021-03-15 at 21 33 15

**Review by Polsonby — APPROVED** [review]
LGTM :) nice

**Pull request merged**